lib/pf/UnifiedApi/Controller/Config.pm

```diff
@@ -327,6 +334,351 @@ sub sort_items {
     return $self->render(json => {});
 }
 
+=head2 options
+
+Handle the option HTTP method
+
+=cut
+
+sub options {
+    my ($self) = @_;
+    my $form = $self->form;
+    return $self->render(json => $self->options_from_form($form));
+}
+
+=head2 options_from_form
+
+Get the options from the form
+
+=cut
+
+sub options_from_form {
+    my ($self, $form) = @_;
+    my (%defaults, %placeholders, %allowed, %meta);
+    my %output = (
+        defaults => \%defaults,
+        placeholders => \%placeholders,
+        allowed => \%allowed,
+        meta => \%meta,
+    );
+
+    my $defaultValues = $self->default_values;
+    for my $field ($form->fields) {
+        my $name = $field->name;
+        $defaults{$name} = $self->field_default($field, $defaultValues);
+        $placeholders{$name} = $self->field_placeholder($field);
+        $allowed{$name} = $self->field_allowed($field);
+        $meta{$name} = $self->field_meta($field);
+    }
+
+    return \%output;
+}
+
+=head2 field_meta
+
+Get a field's meta data
+
+=cut
+
+sub field_meta {
+    my ($self, $field, $no_array) = @_;
+    my $type = $self->field_type($field, $no_array);
+    return {
+        type        => $type,
+        required    => $self->field_is_required($field),
+        placeholder => $self->field_placeholder($field),
+        default     => $self->field_default($field),
+        allowed     => $self->field_allowed($field),
+        $self->field_extra_meta($field, $type),
+    };
+}
+
+=head2 field_extra_meta
+
+Get the extra meta data for a field
+
+=cut
+
+sub field_extra_meta {
+    my ($self, $field, $type) = @_;
+    my %extra;
+    if ($type eq 'array') {
+        $extra{item} = $self->field_meta_array_items($field, 1);
+    } elsif ($type eq 'object') {
+        $extra{properties} = $self->field_meta_object_properties($field);
+    } else {
+        if ($field->isa("HTML::FormHandler::Field::Text")) {
+            $self->field_text_meta($field, \%extra);
+        }
+
+        if ($field->isa("HTML::FormHandler::Field::Integer") || $field->isa("HTML::FormHandler::Field::IntRange")) {
+            $self->field_integer_meta($field, \%extra);
+        }
+    }
+
+    return %extra;
+}
+
+=head2 field_meta_object_properties
+
+Get the properties of a field
+
+=cut
+
+sub field_meta_object_properties {
+    my ($self, $field) = @_;
+    my %p;
+    for my $f ($field->fields) {
+        $p{$f->name} = $self->field_meta($f);
+    }
+
+    return \%p;
+}
+
+=head2 field_integer_meta
+
+Update integer field meta data
+
+=cut
+
+sub field_integer_meta {
+    my ($self, $field, $extra) = @_;
+    my $min = $field->range_start;
+    my $max = $field->range_end;
+    if (defined $min) {
+        $extra->{min_value} = $min;
+    } elsif ($field->isa("HTML::FormHandler::Field::PosInteger")) {
+        $extra->{min_value} = 0;
+    }
+
+    if (defined $max) {
+        $extra->{max_value} = $max;
+    }
+
+    return ;
```

Comment:
Might as well just remove this if we're not returning anything
Unless there is some perl magic this does that I don't get

Reply:
That is to avoid perl from returning the value of the last statement